File: Data Structures & Algorithms/concatenation-of-array/submission-0.cs
public class Solution {
    public int[] GetConcatenation(int[] nums) {
        int n = nums.Length;
        int[] result = new int[2 * n];
        for (int i = 0; i < n; i++)
        {
            result[i] = nums[i];
            result[i + n] = nums[i];
        }

        return result;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> result = new ArrayList<>();
        backtrack(nums, target, 0, new ArrayList<>(), result);
        return result;
    }

    private void backtrack(int[] nums, int remaining, int start, List<Integer> current, List<List<Integer>> result) {
        if (remaining == 0) {
            result.add(new ArrayList<>(current));
            return;
        }

        if (remaining < 0) return; // pruning because current sum already exceed the target

        for (int i = start; i < nums.length; i++) {
            current.add(nums[i]);
            backtrack(nums, remaining - nums[i], i, current, result); // start = i instead of i + 1 => allow reuse
            current.remove(current.size() - 1); // backtrack
        }
    }
}
